File: code/functions/pyprism/pyprism/data/tests/test_load_ids.py
# -*- coding: utf-8 -*-
"""
@created: Apr 15 2021
@modified: Apr 15 2021
@author: Yoann Pradat

Tests for _load_ids.py module.
"""

import pandas as pd
from .._load_ids import load_ids

# User functions =======================================================================================================

def test_load_ids():
    df_ids = load_ids(study="met500")
    assert type(df_ids) == pd.core.frame.DataFrame

    df_ids = load_ids(study="prism")
    assert type(df_ids) == pd.core.frame.DataFrame

    df_ids = load_ids(study="tcga")
    assert type(df_ids) == pd.core.frame.DataFrame

File: code/functions/rprism/tests/testthat/test-load_ids.R
# MET500

test_that("load_ids() met500 all works", {
  skip_on_cran()

  cat("\n")
  df_ids <- load_ids(study="met500")
  expect_true(is(df_ids, "data.frame"))

})


# PRISM

test_that("load_ids() prism all works", {
  skip_on_cran()

  cat("\n")
  df_ids <- load_ids(study="prism")
  expect_true(is(df_ids, "data.frame"))

})


# TCGA

test_that("load_ids() tcga all works", {
  skip_on_cran()

  cat("\n")
  df_ids <- load_ids(study="tcga")
  expect_true(is(df_ids, "data.frame"))

})
